Add reference files for end-to-end testing

arunkannawadi opened the pull request.

arunkannawadi committed:
Add reference files for end-to-end testing

sidneymau commented:
From @aguinot:

> Objects are drawn in order so you will draw only the first 10 objects in the list. Given that stars are at the end, you will not test some particular case regarding the PSF handling. If you want to limit the number of objects drawn you shoudl make sure that they are representative enough of all the cases.
> Maybe a way to do that would be link this to a coverage test so we can check if critical part of the code are not covered by the test

sidneymau committed:
Update config

Specify `image.nobjects: 10` so it is lighter to run in GitHub actions

Change output format from .fits.gz to .fits so that sha can be used to compare

Commit sha from updated hack output

sidneymau commented:
N.B. `hack.sha` produced by
```
find output -type f -exec sha256sum {} \; > hack.sha
```
and you can validate new outputs with `sha256sum -c hack.sha` (will require dir format to be the same)

aguinot commented:
> From @aguinot:
> 
> > Objects are drawn in order so you will draw only the first 10 objects in the list. Given that stars are at the end, you will not test some particular case regarding the PSF handling. If you want to limit the number of objects drawn you shoudl make sure that they are representative enough of all the cases.
> > Maybe a way to do that would be link this to a coverage test so we can check if critical part of the code are not covered by the test

Actually, there is another problem by limiting the number of objects particularly to small numbers. Given that there is a buffer defined around the image, you could end up in case where none of the objects are drawn.

sidneymau committed:
track reference

sidneymau committed:
add reference (via lfs)

sidneymau committed:
use gz compression

sidneymau committed:
revert to full hack.yaml run

sidneymau committed:
draw with photon shooting

sidneymau committed:
upd hack

sidneymau commented:
@arunkannawadi do we want to go ahead and merge this to `main` so that we can start to track new changes to the end-to-end test more carefully?

sidneymau reviewed (approved):
ready to be merged pending confirmation

arunkannawadi commented:
Approved
